contracts/contracts/strategies/ThreePoolStrategy.sol

```diff
@@ -0,0 +1,256 @@
+pragma solidity 0.5.11;
+
+/**
+ * @title Curve 3Pool Strategy
+ * @notice Investment strategy for investing stablecoins via Curve 3Pool
+ * @author Origin Protocol Inc
+ */
+import { ICurvePool } from "./ICurvePool.sol";
+import { ICurveGauge } from "./ICurveGauge.sol";
+import { ICRVMinter } from "./ICRVMinter.sol";
+import {
+    IERC20,
+    InitializableAbstractStrategy
+} from "../utils/InitializableAbstractStrategy.sol";
+
+contract ThreePoolStrategy is InitializableAbstractStrategy {
+    event RewardTokenCollected(address recipient, uint256 amount);
+
+    address crvGaugeAddress;
+    address crvMinterAddress;
+    int128 poolCoinIndex = -1;
+
+    /**
+     * Initializer for setting up strategy internal state. This overrides the
+     * InitializableAbstractStrategy initializer as Curve strategies don't fit
+     * well within that abstraction.
+     * @param _platformAddress Address of the Curve 3pool
+     * @param _vaultAddress Address of the vault
+     * @param _rewardTokenAddress Address of CRV
+     * @param _asset Address of the supported asset
+     * @param _pToken Correspond platform token addres (i.e. 3Crv)
+     * @param _crvGaugeAddress Address of the Curve DAO gauge for this pool
+     * @param _crvMinterAddress Address of the CRV minter for rewards
+     */
+    function initialize(
+        address _platformAddress, // 3Pool address
+        address _vaultAddress,
+        address _rewardTokenAddress, // CRV
+        address _asset,
+        address _pToken,
+        address _crvGaugeAddress,
+        address _crvMinterAddress
+    ) external onlyGovernor initializer {
+        ICurvePool threePool = ICurvePool(_platformAddress);
+        for (int128 i = 0; i < 3; i++) {
+            if (threePool.coins(uint256(i)) == _asset) poolCoinIndex = i;
+        }
+        require(poolCoinIndex != -1, "Invalid 3pool asset");
+        crvGaugeAddress = _crvGaugeAddress;
+        crvMinterAddress = _crvMinterAddress;
+        InitializableAbstractStrategy._initialize(
+            _platformAddress,
+            _vaultAddress,
+            _rewardTokenAddress,
+            _asset,
+            _pToken
+        );
+    }
+
+    /**
+     * @dev Collect accumulated CRV and send to Vault.
+     */
+    function collectRewardToken() external onlyVault {
+        ICRVMinter minter = ICRVMinter(crvMinterAddress);
+        minter.mint(crvGaugeAddress);
+        IERC20 crvToken = IERC20(rewardTokenAddress);
+        uint256 balance = crvToken.balanceOf(address(this));
+        require(
+            crvToken.transfer(vaultAddress, balance),
+            "Reward token transfer failed"
+        );
+        emit RewardTokenCollected(vaultAddress, balance);
```

Comment:
small things: 

1) wrt transfer, maybe can use safeTransfer since you're putting a require on it anyway 
2) anyway the erc20 will emit the transfer, so what's the extra event benefit